From c0fd33a195288fe8bc709f219af1fea892b1c155 Mon Sep 17 00:00:00 2001
From: Ilies CHERGUI <ichergui@nvidia.com>
Date: Thu, 15 Jan 2026 14:46:40 -0800
Subject: [PATCH] Makefile: update for OE builds

* Pass needed make variables through from caller
  for the compiler, linker, etc.

* Remove the extra copying of kernel and OOT sources
  where possible (conftest still needs copying, but
  that's small).

* Add KCFLAGS setting to the nvidia-oot make invocation
  to prevent some compilation failures due gcc warning-to-error
  changes in later versions of the compiler

* Fix build warning on rtl8852ce driver

Upstream-Status: Pending

Signed-off-by: Matt Madison <matt@madison.systems>
Signed-off-by: Ilies CHERGUI <ichergui@nvidia.com>
---
 Makefile                                      | 78 +++++++++++--------
 .../phy/rf/halrf_8852c/halrf_ops_rtl8852c.h   |  8 +-
 2 files changed, 49 insertions(+), 37 deletions(-)

diff --git a/Makefile b/Makefile
index 24ee6c5..b915c67 100644
--- a/Makefile
+++ b/Makefile
@@ -2,11 +2,20 @@
 # SPDX-License-Identifier: BSD-3-Clause
 
 KERNEL_HEADERS ?= /lib/modules/$(shell uname -r)/build
-KERNEL_OUTPUT ?= ${KERNEL_HEADERS}
+KERNEL_OUTPUT ?= $(KERNEL_HEADERS)
 
 MAKEFILE_DIR := $(abspath $(shell dirname $(lastword $(MAKEFILE_LIST))))
-NVIDIA_CONFTEST ?= ${MAKEFILE_DIR}/out/nvidia-conftest
-NVIDIA_DTS_BUILD_SCRIPTS ?= $(realpath ${MAKEFILE_DIR}/kernel-devicetree)
+NVIDIA_CONFTEST ?= $(MAKEFILE_DIR)/out/nvidia-conftest
+NVIDIA_DTS_BUILD_SCRIPTS ?= $(realpath $(MAKEFILE_DIR)/kernel-devicetree)
+HOSTCC ?= gcc
+MODLIB ?= /lib/modules
+CC ?= $(CROSS_COMPILE)gcc
+CXX ?= $(CROSS_COMPILE)g++
+LD ?= $(CROSS_COMPILE)ld.bfd
+AR ?= $(CROSS_COMPILE)ar
+OBJCOPY ?= $(CROSS_COMPILE)objcopy
+
+V ?= 0
 
 OPENRM ?= 1
 ifeq ($(OPENRM), 1)
@@ -19,11 +28,6 @@ ifneq ($(words $(subst :, ,$(MAKEFILE_DIR))), 1)
 $(error source directory cannot contain spaces or colons)
 endif
 
-NPROC ?= $(shell nproc)
-
-ifeq ("$(wildcard $(KERNEL_OUTPUT))","")
-$(error kernel headers/output directory "$(KERNEL_OUTPUT)" does not exist!)
-endif
 
 .PHONY : help modules modules_install clean conftest hwpm nvidia-oot nvgpu
 
@@ -42,6 +46,8 @@ dtbs: nvidia-dtbs
 modules_install: hwpm nvidia-oot nvgpu nvidia-display-install
 clean: hwpm nvidia-oot nvgpu nvidia-display-clean nvidia-dtbs-clean conftest-clean
 
+$(NVIDIA_CONFTEST):
+	mkdir -p $@
 
 conftest:
 ifeq ($(MAKECMDGOALS), modules)
@@ -49,12 +55,12 @@ ifeq ($(MAKECMDGOALS), modules)
 	@echo   "make $(MAKECMDGOALS) - conftest ..."
 	@echo   "================================================================================"
 	mkdir -p $(NVIDIA_CONFTEST)/nvidia;
-	cp -av $(MAKEFILE_DIR)/nvidia-oot/scripts/conftest/* $(NVIDIA_CONFTEST)/nvidia/;
-	$(MAKE) -j $(NPROC) ARCH=arm64 \
+	cp -av $(MAKEFILE_DIR)/nvidia-oot/scripts/conftest/* $(NVIDIA_CONFTEST)/nvidia/
+	$(MAKE) ARCH=arm64 \
 		src=$(NVIDIA_CONFTEST)/nvidia obj=$(NVIDIA_CONFTEST)/nvidia \
-		CC=$(CROSS_COMPILE)gcc LD=$(CROSS_COMPILE)ld \
-		NV_KERNEL_SOURCES=$(KERNEL_HEADERS) \
-		NV_KERNEL_OUTPUT=$(KERNEL_OUTPUT) \
+		CC="$(CC)" LD="$(LD)" \
+		NV_KERNEL_SOURCES=$(KERNEL_SRC) \
+		NV_KERNEL_OUTPUT=$(KBUILD_OUTPUT) \
 		-f $(NVIDIA_CONFTEST)/nvidia/Makefile
 endif
 
@@ -66,9 +72,10 @@ hwpm: conftest
 	@echo   "================================================================================"
 	@echo   "make $(MAKECMDGOALS) - hwpm ..."
 	@echo   "================================================================================"
-	$(MAKE) -j $(NPROC) ARCH=arm64 \
-		-C $(KERNEL_OUTPUT) \
+	$(MAKE) ARCH=arm64 \
+		-C $(KERNEL_PATH) \
 		M=$(MAKEFILE_DIR)/hwpm/drivers/tegra/hwpm \
+		MODLIB=$(MODLIB) \
 		CONFIG_TEGRA_OOT_MODULE=m \
 		srctree.hwpm=$(MAKEFILE_DIR)/hwpm \
 		srctree.nvconftest=$(NVIDIA_CONFTEST) \
@@ -82,10 +89,12 @@ nvidia-oot: conftest hwpm
 	@echo   "================================================================================"
 	@echo   "make $(MAKECMDGOALS) - nvidia-oot ..."
 	@echo   "================================================================================"
-	$(MAKE) -j $(NPROC) ARCH=arm64 \
-		-C $(KERNEL_OUTPUT) \
+	$(MAKE) ARCH=arm64 \
+		-C $(KERNEL_PATH) \
 		M=$(MAKEFILE_DIR)/nvidia-oot \
+		MODLIB=$(MODLIB) \
 		CONFIG_TEGRA_OOT_MODULE=m \
+		KCFLAGS="-Wno-error=address" \
 		srctree.nvidia-oot=$(MAKEFILE_DIR)/nvidia-oot \
 		srctree.hwpm=$(MAKEFILE_DIR)/hwpm \
 		srctree.nvconftest=$(NVIDIA_CONFTEST) \
@@ -105,9 +114,10 @@ nvgpu: conftest nvidia-oot
 	@echo   "================================================================================"
 	@echo   "make $(MAKECMDGOALS) - nvgpu ..."
 	@echo   "================================================================================"
-	$(MAKE) -j $(NPROC) ARCH=arm64 \
-		-C $(KERNEL_OUTPUT) \
+	$(MAKE) ARCH=arm64 \
+		-C $(KERNEL_PATH) \
 		M=$(MAKEFILE_DIR)/nvgpu/drivers/gpu/nvgpu \
+		MODLIB=$(MODLIB) \
 		CONFIG_TEGRA_OOT_MODULE=m \
 		srctree.nvidia=$(MAKEFILE_DIR)/nvidia-oot \
 		srctree.nvidia-oot=$(MAKEFILE_DIR)/nvidia-oot \
@@ -117,19 +127,19 @@ nvgpu: conftest nvidia-oot
 endif
 
 define display-cmd
-	$(MAKE) -j $(NPROC) ARCH=arm64 TARGET_ARCH=aarch64 \
+	$(MAKE) ARCH=arm64 TARGET_ARCH=aarch64 \
 		-C $(MAKEFILE_DIR)/$(NVIDIA_DISPLAY_DIR) \
-		LOCALVERSION=$(version) \
-		NV_VERBOSE=0 \
+		NV_VERBOSE=$(V) \
 		KERNELRELEASE="" \
 		SYSSRCNVOOT=$(MAKEFILE_DIR)/nvidia-oot \
-		SYSSRC=$(KERNEL_HEADERS) \
-		SYSOUT=$(KERNEL_OUTPUT) \
-		CC=$(CROSS_COMPILE)gcc \
-		LD=$(CROSS_COMPILE)ld.bfd \
-		AR=$(CROSS_COMPILE)ar \
-		CXX=$(CROSS_COMPILE)g++ \
-		OBJCOPY=$(CROSS_COMPILE)objcopy
+		SYSSRC=$(KERNEL_SRC) \
+		SYSOUT=$(KBUILD_OUTPUT) \
+		KBUILD_EXTRA_SYMBOLS=$(MAKEFILE_DIR)/nvidia-oot/Module.symvers \
+		CC="$(CC)" \
+		LD="$(LD)" \
+		AR="$(AR)" \
+		CXX="$(CXX)" \
+		OBJCOPY="$(OBJCOPY)"
 endef
 
 
@@ -151,7 +161,9 @@ nvidia-display-install:
 	@echo   "================================================================================"
 	@echo   "make $(MAKECMDGOALS) - nvidia-display ..."
 	@echo   "================================================================================"
-	$(MAKE) -C $(KERNEL_OUTPUT) INSTALL_MOD_DIR=updates/opensrc-disp \
+	$(MAKE) -C $(KERNEL_PATH) \
+		KERNEL_MODLIB=$(MODLIB) \
+		INSTALL_MOD_DIR=updates/opensrc-disp \
 		M=$(MAKEFILE_DIR)/$(NVIDIA_DISPLAY_DIR)/kernel-open modules_install
 
 nvidia-display-clean:
@@ -169,10 +181,10 @@ nvidia-dtbs:
 	@echo   "make nvidia-dtbs ..."
 	@echo   "================================================================================"
 	TEGRA_TOP=$(MAKEFILE_DIR) \
-	srctree=$(KERNEL_HEADERS) \
-	objtree=$(KERNEL_OUTPUT) \
+	srctree=$(KERNEL_SRC) \
+	objtree=$(KBUILD_OUTPUT) \
 	oottree=$(NVIDIA_DTS_BUILD_SCRIPTS) \
-	HOSTCC=gcc \
+	HOSTCC="$(HOSTCC)" \
 	$(MAKE) -f $(NVIDIA_DTS_BUILD_SCRIPTS)/scripts/Makefile.build \
 		obj=$(NVIDIA_DTS_BUILD_SCRIPTS)/generic-dts \
 		dtbs
diff --git a/nvidia-oot/drivers/net/wireless/realtek/rtl8852ce/phl/hal_g6/phy/rf/halrf_8852c/halrf_ops_rtl8852c.h b/nvidia-oot/drivers/net/wireless/realtek/rtl8852ce/phl/hal_g6/phy/rf/halrf_8852c/halrf_ops_rtl8852c.h
index 7966290..b4b92ad 100644
--- a/nvidia-oot/drivers/net/wireless/realtek/rtl8852ce/phl/hal_g6/phy/rf/halrf_8852c/halrf_ops_rtl8852c.h
+++ b/nvidia-oot/drivers/net/wireless/realtek/rtl8852ce/phl/hal_g6/phy/rf/halrf_8852c/halrf_ops_rtl8852c.h
@@ -22,13 +22,13 @@
  * Larry Finger <Larry.Finger@lwfinger.net>
  *
  *****************************************************************************/
-#ifndef __HALRF_OPS_RTL8852C_H__
-#define __HALRF_OPS_RTL8852C__
+#ifndef HALRF_OPS_RTL8852C_H
+#define HALRF_OPS_RTL8852C_H
 #ifdef RF_8852C_SUPPORT
 
 
 void rf_set_ops_8852c(struct rf_info *rf) ;
 
 
-#endif
-#endif /*  __HALRF_OPS_RTL8852C_H__ */
+#endif /* RF_8852C_SUPPORT */
+#endif /* HALRF_OPS_RTL8852C_H */
-- 
2.34.1

